nuxt.config.js

```diff
@@ -5,27 +5,37 @@ export default {
 	mode: "universal",
 	generate: {
 		fallback: true,
-		routes (callback) {
+		async routes (callback) {
+			// Determine how many pages of posts are available
+			const totalPages = await (await axios.get(`${Config.wpDomain}${Config.apiBase}posts`)).headers["x-wp-totalpages"]
+			// Create empty array to hold all retrieved post data in chunks of 10
+			const postPagesAPI = []
+			// Create empty array to hold all pagination routes for the News and Views page (e.g. /page/1, /page/2, etc)
+			const newsPaginationRoutes = []
+			for (let postPage = 1; postPage <= totalPages; postPage++) {
+				// Add each page of posts to postPagesAPI
+				postPagesAPI.push(axios.get(`${Config.wpDomain}${Config.apiBase}posts?page=${postPage}`))
+				// Add each page index to newsPaginationRoutes
+				newsPaginationRoutes.push({ route: `/news-and-views/page/${postPage}`, payload: postPage })
+			}
+
+			// Spread the entries in postPagesAPI and fetch each post
 			axios.all([
-				axios.get(`${Config.wpDomain}${Config.apiBase}pages`),
-				axios.get(`${Config.wpDomain}${Config.apiBase}posts`)
+				...postPagesAPI
 			])
-				.then(axios.spread(function (pages, posts) {
-					const pageRoutes = pages.data.map((page) => {
-						return {
-							route: (page.slug !== "home") ? "/" + page.slug : "/",
-							payload: page
-						}
-					})
-
+				.then(axios.spread((posts) => {
+					// Add each post to the postRoutes variable
+					// replace line below with this when totalPages > 1:
+					// const postRoutes = posts.data.map((...post) => {
```

Comment:
This needs to be done for this pull request to be complete.

Reply:
What needs to be done?

Reply:
if you change `posts` to `...post` and `totalPages` <= 1 the code breaks

Reply:
Can you write code that handles both situations? It will need to work no matter how many pages of results there are.